### ISSUE TITLE
Implement replay tracking system for testing [Blocked: #59]

### ISSUE THREAD

**Issue by BenHenning** [oppia/oppia-android#101]
We should consider implementing a tester-only tracking system for user actions that can be replayed. If we do this exhaustively (maybe via a tracing system) and robustly (such as ensuring recent actions during a crash are saved via a crash handler) we may be able to replay a tester’s experience verbatim to hit the same broken state they observed, allowing for much quicker bug fixes. 

This can be coupled with a content observer dump method to include the replay actions in an Android bug report. We should also have the ability to emulate networking and file I/O when running the app to avoid needing real authentication or saving actual progress.

**Comment by BenHenning** [comment]
Targeting MVP as a nice-to-have, but if we can build it sooner (e.g. for prototype) it can help play a significant role in quickly stabilizing the app.

**Comment by BenHenning** [comment]
Also, this is blocked on #59 for privacy reasons: we must make sure that such a tracking system is not accidentally shipped on prod versions of the app.

**Comment by EskAlam06** [comment]
this should remain open , as it's enhancement related
